### DISCUSSION TITLE
FLTK implementation

### DISCUSSION THREAD

**Discussion by mgord9518** [tsujan/Kvantum#828] [General]
Hello, I've used Kvantum for pretty much every Qt-based desktop I've ever installed and it's always been great. I've recently gotten into Zig and FLTK and have considered porting at least a subset of Kvantum over to its theming engine to allow Kvantum themes to be used there too.

I've read through the docs and have a basic plan as to how to accomplish this as the main parts seem to be pretty simple (load SVG, find whatever shape/group of the widget should be drawn then finally stretch it, rasterize and draw) so I'm just wondering if I'm missing anything big. All the more advanced features will obviously be much more tedious to implement, but the basics should be enough to make most apps feel native (I think?)

As for why I don't just want to use Qt, I think its a fantastic framework but I just find FLTK more fun to use and it being much lighter weight is a plus.

**Comment by tsujan** [comment]
I can't comment on FLTK because my knowledge of it is minimal.

As for Qt, using of SVG elements for drawing widgets seemed straightforward at first, 9 years ago, and that was why I started Kvantum. But, if I knew how difficult the real job would be, I might never start it ;) Fortunately, all obstacles have been overcome, but it was a very long and complex process.

I think it depends on the toolkit.

**Reply by mgord9518** [reply]
Thanks for the quick reply! Everything's always a lot easier in theory than practice lmao, I guess I'll start messing around with it and post some updates in this thread if it gets anywhere
